redis/main.tf

```diff
@@ -1,60 +1,61 @@
 resource "aws_elasticache_replication_group" "redis" {
-  count                         = "${var.auth_token == "" ? 1 : 0}"
+  count                         = var.auth_token == "" ? 1 : 0
   replication_group_id          = "${var.project}-${var.environment}-${var.name}"
   replication_group_description = "Redis cluster for ${var.project}-${var.environment}-${var.name}"
   engine                        = "redis"
-  engine_version                = "${var.engine_version}"
-  port                          = "${var.port}"
-  node_type                     = "${var.node_type}"
-  number_cache_clusters         = "${var.num_cache_nodes}"
-  parameter_group_name          = "${var.parameter_group_name}"
-  security_group_ids            = ["${aws_security_group.sg_redis.id}"]
-  subnet_group_name             = "${aws_elasticache_subnet_group.elasticache.id}"
-  availability_zones            = ["${var.availability_zones}"]
-  automatic_failover_enabled    = "${var.automatic_failover_enabled}"
-  snapshot_window               = "${var.snapshot_window}"
-  snapshot_retention_limit      = "${var.snapshot_retention_limit}"
-  snapshot_arns                 = "${var.snapshot_arns}"
-  at_rest_encryption_enabled    = "${var.at_rest_encryption_enabled}"
-  transit_encryption_enabled    = "${var.transit_encryption_enabled}"
+  engine_version                = var.engine_version
+  port                          = var.port
+  node_type                     = var.node_type
+  number_cache_clusters         = var.num_cache_nodes
+  parameter_group_name          = var.parameter_group_name
+  security_group_ids            = [aws_security_group.sg_redis.id]
+  subnet_group_name             = aws_elasticache_subnet_group.elasticache.id
+  availability_zones            = var.availability_zones
+  automatic_failover_enabled    = var.automatic_failover_enabled
+  snapshot_window               = var.snapshot_window
+  snapshot_retention_limit      = var.snapshot_retention_limit
+  snapshot_arns                 = var.snapshot_arns
+  at_rest_encryption_enabled    = var.at_rest_encryption_enabled
+  transit_encryption_enabled    = var.transit_encryption_enabled
 
-  tags {
+  tags = {
     Name        = "${var.project}-${var.environment}-${var.name}"
-    Environment = "${var.environment}"
-    Project     = "${var.name}"
+    Environment = var.environment
+    Project     = var.name
   }
 }
 
 resource "aws_elasticache_replication_group" "redis_auth" {
```

Comment:
You should be able to simplify the module now by using `null` as default for the var.auth_token.

Reply:
There'll be a terraform state move needed for current setups that use auth. I'd make sure to document that in the release notes